server/routes/csv/csv-router.js

```diff
@@ -2,27 +2,32 @@ import fastcsv from 'fast-csv';
 import fs from 'fs';
 import express from 'express';
 import AppError from '../../errors/AppError.js';
-
-import getAllTreeDataByCity from './csv-queries.js';
+import { getAllTreeDataByCity, getAllTreeDataCities } from './csv-queries.js';
 
 const csvRouter = express.Router();
 
 csvRouter.get('/', async (req, res) => {
   const { city } = req.query;
-  const data = await getAllTreeDataByCity(city);
+  const data = city
+    ? await getAllTreeDataByCity(city)
+    : await getAllTreeDataCities();
 
-  if (!data) {
+  if (!data || data === undefined || data.length === 0) {
     throw new AppError(404, 'Failed to find any data.');
   }
 
   const jsonData = JSON.parse(JSON.stringify(data));
-  const cityPath = `server/csv-downloads/${city.toLowerCase()}.csv`;
-  const ws = fs.createWriteStream(cityPath);
+  const cityName = city
+    ? city.toLowerCase().replaceAll(' ', '_')
```

Comment:
Do we want to replace with a `_` or `-` to be consistent with the `all-cities` below?